config/nrfconnect/chip-module/Kconfig

```
@@ -70,3 +70,145 @@ config CHIP_DEBUG_SYMBOLS
 	default y
 	help
 	  Build the application with debug symbols.
+
+# Factory data
+config CHIP_FACTORY_DATA_ENABLE
+	bool "Enables Factory Data support"
+	default n
+	help
+		Enables generating a factory data after build.
+		To generate factory data the nrfconnect_generate_factory_data function from
+		generate_factory_data.cmake should be called.
+		In result a new output files will be created:
+		- merged_fd.hex file containing firmware and factory data merged
+		- factory_data.hex file containing only a factory data with proper memory offset.
+  
+config CHIP_FACTORY_DATA_PARTITION
+	hex "Localization in NVS storage where Factory Data partition will be stored"
+	depends on CHIP_FACTORY_DATA_ENABLE
+	default 0xfb000
+	help 
+		FLASH Memory location where created factory data will be stored.
+
+config CHIP_FACTORY_DATA_PARTITION_SIZE
+	hex "Maximum size of partition data"
+	depends on CHIP_FACTORY_DATA_ENABLE
+	default 0x1000
+	help
+		Provide maximum partition size to verify that 
```

I would simplify this description to the similar as above. User don't really care what purpose this size will be used in our script, so maybe it is too verbose.
```suggestion
		The size of factory data partition located in a flash memory.
```